diag_manager/fms_diag_output_buffer.F90

```diff
@@ -331,6 +341,24 @@ subroutine set_yaml_id(this, yaml_id)
   this%yaml_id = yaml_id
 end subroutine set_yaml_id
 
+!> @brief Determine if finished with math
+!! @return this%done_with_math
+function is_done_with_math(this) &
+  result(res)
+  class(fmsDiagOutputBuffer_type), intent(in) :: this        !< Buffer object
+  logical :: res
+
+  res = this%done_with_math
+end function is_done_with_math
+
+!> @brief Set done_with_math to .true.
+subroutine set_done_with_math(this)
```

Comment:
I don't think you will you ever set it back to .false. 
This will be set to .true. for variables that are static after the "math" (a copy) is done for the first time.